src/routes/Attendance.ts

```diff
@@ -0,0 +1,299 @@
+import StatusCodes from "http-status-codes";
+import e, { Request, Response } from "express";
+import { join } from "path";
+import fs from "fs";
+import logger from "@shared/Logger";
+import path from "path/posix";
+import util from "util";
+
+import { FieldSet, Records } from "airtable";
+
+import StudentDao from "@daos/Airtable/StudentDao";
+import CanvasCoursesDao from "@daos/Airtable/CanvasCoursesDao";
+import ModulesDao from "@daos/Canvas/ModulesDao";
+import { paramMissingError } from "@shared/constants";
+import SubmissionDao from "@daos/Canvas/SubmissionDao";
+import Module from "@entities/Module";
+import Submission from "@entities/Submission";
+import ModuleItem from "@entities/ModuleItem";
+
+const studentDao = new StudentDao();
+const canvasCoursesDao = new CanvasCoursesDao();
+const modulesDao = new ModulesDao();
+const submissionDao = new SubmissionDao();
+
+const { BAD_REQUEST, CREATED, OK } = StatusCodes;
+
+interface IMilestone {
+  name: string;
+  itemId: number;
+  points: number;
+  completed: boolean;
+}
+
+/**
+ * Get a set of learners' student records from Airtable by their email.
+ *
+ * @param learners
+ * @returns
+ */
+async function getStudentRecords(
+  learners: Record<string, unknown>[]
+): Promise<Records<FieldSet>> {
+  const emails: string[] = learners.map((learner) => learner.Email as string);
+  return await studentDao.getByEmails(emails);
+}
+
+/**
+ * Merge each learner's Lambda ID and Labs Role from their student record.
+ * (Mutates learners array.)
+ *
+ * @param learners
+ * @returns
+ */
+function mergeStudentRecords(
```

Comment:
Looks like it's near time to create domain objects for specific collections of functions.

Reply:
Starting to get confused between DDD and service architecture—would domain objects mirror the route structure?

Reply:
[Added a service layer](https://github.com/Lambda-School-Labs/labs-be/pull/14/commits/cb9e0b0acfeec828a74d8196bbb03d573832b01c)

Reply:
No, a domain object would contain functions that operate on a given domain. Ie, Canvas object could be a full domain and all the specific logic wanted would go into it. That would likely get huge, so having a `domains\canvas` folder with sub domain classes makes more sense. Routers would use these objects.

It's gets hairy when the domain boundaries are crossed. Ie. You shouldn't use an Airtable domain object inside of a canvas class. It happens tho. At first it makes sense to get things going, but over time it's a huge mess.

Let's get this out first then we can refactor into domains.